cg/server/api.py

```diff
@@ -278,13 +278,24 @@ def sample_in_collaboration(sample_id):
 @BLUEPRINT.route("/pools")
 def pools():
     """Fetch pools."""
-    customer_objs: Optional[Customer] = (
-        None if g.current_user.is_admin else g.current_user.customers
+    customer_objs: Optional[List[Customer]] = (
```

Comment:
@henrikstranneheim
This is what we slacked about the other day. 

```Python
get_pools_for_customer()
```

had three function outcomes:
1. Return all pools
2. Return pools for customer
3. Return pools for customer where name or order match enquiry

I broke this function up into: 

```Python
get_pools_by_customer_id()
get_pools_by_name_enquiry()
get_pools_by_oder_enquiry()
```
Then I extracted the logic of constructing the end result into 
```servers/api.py```

I think this shows a bit better what is going on under the hood, but is still messy... 
Plus we have query functions that only filter on a single attribute.

I tested the functionality of this construction and it behaves identical to the previous implementation.